src/esbmc/options.cpp

```diff
@@ -548,6 +555,10 @@ const struct group_opt_templ all_cmd_options[] = {
      {"branch-function-coverage-claims",
       NULL,
       "enable branch-coverage-ext and shows all reached claims"},
+     {"assign-param-nondet",
```

Comment:
Isn't it already the case?

Reply:
I feel like we should have an option to control (currently this algorithm is enabled by default when using `--function`). 

The main reason is that this would cause some trouble in solidity, particularly for Object-Type parameters. Given that solidity frontend now based on the cpp-frontend, I would assume this might also affect c++.